packages/components/src/List/Toolbar/ColumnChooserButton/ColumnChooser/ColumnChooser.component.js

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { ColumnChooserProvider } from './columnChooser.context';
+import { useColumnChooserManager } from '../hooks';
+import FilterBar from '../../../../FilterBar';
+import ColumnChooserBody from './ColumnChooserBody';
+import ColumnChooserFooter from './ColumnChooserFooter';
+import ColumnChooserHeader from './ColumnChooserHeader';
+import getDefaultT from '../../../../translate';
+import cssModule from './ColumnChooser.scss';
+import { getTheme } from '../../../../theme';
+
+const theme = getTheme(cssModule);
+
+const haveColumnLabel = label => column => column.label.toLowerCase().includes(label.toLowerCase());
+const filterColumnsChooser = (columns, filter) => columns.filter(haveColumnLabel(filter));
+const changeVisibleToHidden = column => ({ ...column, hidden: !column.visible });
+const transformColumnsChooser = columns => columns.map(changeVisibleToHidden);
+
+export default function ColumnChooser({
+	children,
+	columns,
+	filterValue,
+	id,
+	nbLockedLeftItems = 0,
+	onClose,
+	submit,
+	t = getDefaultT(),
+}) {
+	const { columnsChooser, onChangeVisibility, onSelectAll, selectAll } = useColumnChooserManager(
+		columns,
+		nbLockedLeftItems,
+	);
+
+	useEffect(() => {
+		// eslint-disable-next-line no-console
+		console.warn(
+			'Guideline and development of the ColumnChooser component still in progress. It may have breaking change in the future',
+		);
+	}, []);
+
+	// We are transforming back the field visible to hidden, to be compliant with the list.
+	const onSubmit = event => {
+		event.preventDefault();
+		submit(event, transformColumnsChooser(columnsChooser));
+	};
+	const [filter, setFilter] = useState(filterValue || '');
+	const onFilter = (_, value) => setFilter(value);
+	const resetFilter = () => setFilter('');
+	const filteredColumnsChooser = useMemo(() => filterColumnsChooser(columnsChooser, filter), [
+		columnsChooser,
+		filter,
+	]);
+	const Default = (
+		<React.Fragment>
+			<ColumnChooserHeader />
+			<FilterBar
+				autoFocus={false}
+				className={theme('tc-column-chooser-filter')}
+				dockable={false}
+				docked={false}
+				iconAlwaysVisible
+				id={`${id}-filter`}
+				placeholder={t('FIND_COLUMN_FILTER_PLACEHOLDER', {
+					defaultValue: 'Find a column',
+				})}
+				onToggle={resetFilter}
+				onFilter={onFilter}
+				value={filter}
+			/>
+			<form id={`${id}-form`} className={theme('tc-column-chooser')} onSubmit={onSubmit}>
+				<ColumnChooserBody />
+				<ColumnChooserFooter />
+			</form>
+		</React.Fragment>
+	);
+	return (
+		<ColumnChooserProvider
+			value={{
+				columnsChooser: filteredColumnsChooser,
```

Comment:
columns please 🙏 
```suggestion
				columns: filteredColumns,
```

Reply:
I'm ok with columns, np, at a moment I used it. But the shape of a column in the column chooser is not the same as a column in the list.
So i can go back to name everything columns, no problem but I will be confusing with the columns of the list.

Reply:
The ColumnChooserButton and ColumnChooser components waiting for a columns props (the one from the list) and at mounting this columns are transformed (in the hook) to columns chooser shape.
Do you have a naming idea to dissociate the two ?

Reply:
I don't think this is that confusing. You can wait columns from the api, but internally you always have the adapted ones. So you have this adaptation step at the beginning, then everywhere in the column chooser you have the chooser columns right ?